metrics/nist/nist.py

```diff
@@ -0,0 +1,132 @@
+    ...               'ensures', 'that', 'the', 'military', 'always',
+    ...               'obeys', 'the', 'commands', 'of', 'the', 'party']
+
+    >>> reference1 = ['It', 'is', 'a', 'guide', 'to', 'action', 'that',
+    ...               'ensures', 'that', 'the', 'military', 'will', 'forever',
+    ...               'heed', 'Party', 'commands']
+
+    >>> reference2 = ['It', 'is', 'the', 'guiding', 'principle', 'which',
+    ...               'guarantees', 'the', 'military', 'forces', 'always',
+    ...               'being', 'under', 'the', 'command', 'of', 'the',
+    ...               'Party']
+
+    >>> reference3 = ['It', 'is', 'the', 'practical', 'guide', 'for', 'the',
+    ...               'army', 'always', 'to', 'heed', 'the', 'directions',
+    ...               'of', 'the', 'party']
+    
+    >>> nist.compute(hypothesis1, [reference1, reference2])
+    {'nist': 3.3709935957649324}
+"""
+
+
+@evaluate.utils.file_utils.add_start_docstrings(_DESCRIPTION, _KWARGS_DESCRIPTION)
+class NIST(evaluate.Metric):
+    """A wrapper around NLTK's NIST implementation."""
+
+    def _info(self):
+        return evaluate.MetricInfo(
+            module_type="metric",
+            description=_DESCRIPTION,
+            citation=_CITATION,
+            inputs_description=_KWARGS_DESCRIPTION,
+            features=[
+                datasets.Features(
+                    {
+                        "predictions": Sequence(Value("string", id="token"), id="prediction"),
+                        "references": Sequence(Sequence(Value("string", id="token"), id="reference"), id="references"),
+                    }
+                ),
+                datasets.Features(
+                    {
+                        "predictions": Sequence(
+                            Sequence(Value("string", id="token"), id="prediction"), id="predictions"
+                        ),
+                        "references": Sequence(
+                            Sequence(Sequence(Value("string", id="token"), id="reference"), id="references"),
+                            id="reference_corpus",
+                        ),
+                    }
+                ),
+            ],
+            homepage="https://www.nltk.org/api/nltk.translate.nist_score.html",
+            codebase_urls=["https://github.com/nltk/nltk/blob/develop/nltk/translate/nist_score.py"],
+            reference_urls=["https://en.wikipedia.org/wiki/NIST_(metric)"],
+        )
+
+    def _compute(self, predictions, references, n: int = 5):
+        try:
+            if isinstance(predictions[0], str) and isinstance(references[0][0], str):
+                return {"nist": sentence_nist(references=references, hypothesis=predictions, n=n)}
+            elif isinstance(predictions[0][0], str) and isinstance(references[0][0][0], str):
+                return {"nist": corpus_nist(list_of_references=references, hypotheses=predictions, n=n)}
+            else:
+                raise TypeError
```

Comment:
Unless something is broken with the `Features` this should never happen. The data type should be enforced by arrow following the specified formats in `features`.

Reply:
Okay, I will remove the else-clause.